util/ratelimits.go

```diff
@@ -54,29 +54,35 @@ func (c *PostRequestCounter) ApiResponseRatelimitCounter(resp *resty.Response) e
 		return err
 	}
 
-	c.RefreshTime, err = strconv.Atoi(resp.Header().Get("X-Ratelimit-Reset"))
+	epochTime, err := strconv.Atoi(resp.Header().Get("X-Ratelimit-Reset"))
 	if err != nil {
 		return err
 	}
+	c.RefreshTime = time.Unix(int64(epochTime), 0)
+	secondaryRefreshTime := time.Unix(int64(epochTime)-int64(reconciler.SecondaryLinodeTooManyPOSTRequestsErrorRetryDelay.Seconds()), 0)
+
+	// TODO: remove when rate-limits are simplified
+	currTime := time.Now()
+	if c.ReqRemaining >= reconciler.SecondaryPOSTRequestLimit && currTime.Before(secondaryRefreshTime) {
+		c.RefreshTime = secondaryRefreshTime
+	}
```

Comment:
```suggestion
	epochTime, err := strconv.ParseInt(resp.Header().Get("X-Ratelimit-Reset"), 10, 64)
	if err != nil {
		return err
	}
	c.RefreshTime = time.Unix(epochTime, 0)
	// We Add a negative number because for some reason Add returns a Time but Sub returns a Duration...
	secondaryRefreshTime := time.Unix(epochTime, 0).Add(reconciler.SecondaryLinodeTooManyPOSTRequestsErrorRetryDelay * -1)
	// TODO: remove when rate-limits are simplified
	currTime := time.Now()
	if c.ReqRemaining >= reconciler.SecondaryPOSTRequestLimit && currTime.Before(secondaryRefreshTime) {
		c.RefreshTime = secondaryRefreshTime
	}
```
to get rid of some of the int64 casting

Reply:
thanks, updated.